ref/general/securing-microservices.adoc

```diff
@@ -13,33 +13,59 @@
+You can't afford to run a distributed system without a solid, advanced, and reliable monitoring platform.
+To address such challenges, you can use Distributed Tracing.
+Distributed tracing is a method to pinpoint the failures in the system and identify the reason behind it.
+Furthermore, you can also identify the place where failure occurs.
+So, it is very easy to track down, which microservice is facing a security issue.
+Prometheus was built specifically for microservices.
+Prometheus is an open source monitoring platform for microservices and a part of the Cloud Native Computing Foundation.
 
-Microservices communicate with each other through APIs.
-Every point of communication entails security vulnerabilities.
-An API Gateway provides an entry point for client requests and protects the microservices from the client through authentication and authorization.
-Thus, client does not interact with the microservices directly.
+== Cloud-specific vulnerabilities
 
-== Monitoring security and tracing
+The cloud environment makes microservices susceptible to the peculiar security issues affecting cloud, that is, cloud-specific vulnerabilities.
+The Cloud Security Shared Responsibility Model (CSSM), explicitly proposed an approach for resolving these cloud-specific vulnerabilities.
+The CSSM denotes that security providers are responsible for the security of the cloud and customers are responsible for securing the data they put in the cloud. Depending on the type of deployment—IaaS, PaaS, or SaaS—customer responsibilities will be determined.
+While IaaS customers retain a lot of control, they can lean on security providers to manage security from a physical, infrastructure, network, and virtualization standpoint.
+In PaaS, customers focus on deploying and managing applications, as well as managing data, assets, and permissions, security providers take control of operating the underlying infrastructure, including guest operating systems.
+SaaS places the most responsibility on the security providers.
+With the security providers managing the entire infrastructure as well as the applications, customers are only responsible for managing data, as well as user access/identity permissions.
 
-Monitoring microservices is complex, and end-to-end monitoring  is important.
-Prometheus is an open source monitoring platform for microservices and a part of the Cloud Native Computing Foundation.
-In the distributed tracing method of monitoring microservices, a unique ID is generated at login, which helps in tracing and troubleshooting the transaction.
+== Technological diversity concerns
+
+Microservices are built with different business capabilities by different development teams, which may use different technologies i.e. different programming languages and frameworks.
+The motivation for this approach is to use the best tool for specific problems.
+While this feature aids in productivity, it complicates security.
```

Comment:
Change
"While this feature aids in productivity, it complicates security."
To
"While this aspect of microservice development aids in productivity, it complicates security."